phidgets_api/include/phidgets_api/stepper.hpp

```diff
@@ -0,0 +1,116 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef PHIDGETS_API_STEPPER_H
+#define PHIDGETS_API_STEPPER_H
+
+#include <functional>
+
+#include <libphidget22/phidget22.h>
+
+#include "phidgets_api/phidget22.hpp"
+
+namespace phidgets {
+
+class Stepper final
+{
+  public:
+    PHIDGET22_NO_COPY_NO_MOVE_NO_ASSIGN(Stepper)
+
+    explicit Stepper(int32_t serial_number, int hub_port,
+                     bool is_hub_port_device, int channel,
+                     std::function<void(int, double)> position_change_handler,
+                     std::function<void(int, double)> velocity_change_handler,
+                     std::function<void(int)> stopped_handler);
+
+    ~Stepper();
+
+    int32_t getSerialNumber() const noexcept;
+
+    double getAcceleration() const;
+    void setAcceleration(double acceleration) const;
+    double getMinAcceleration() const;
+    double getMaxAcceleration() const;
+    bool stepControlMode() const;
+    void setStepControlMode(bool step_control_mode) const;
+    double getCurrentLimit() const;
+    void setCurrentLimit(double current_limit) const;
+    double getMinCurrentLimit() const;
+    double getMaxCurrentLimit() const;
+    uint32_t getDataInterval() const;
+    void setDataInterval(uint32_t data_interval_ms) const;
+    uint32_t getMinDataInterval() const;
+    uint32_t getMaxDataInterval() const;
+    bool getEngaged() const;
+    void setEngaged(bool engaged) const;
+    void enableFailsafe(uint32_t failsafe_time_ms) const;
+    uint32_t getMinFailsafeTime() const;
+    uint32_t getMaxFailsafeTime() const;
+    double getHoldingCurrentLimit() const;
+    void setHoldingCurrentLimit(double holding_current_limit) const;
+    bool isMoving() const;
+    double getPosition() const;
+    double getMinPosition() const;
+    double getMaxPosition() const;
+    void addPositionOffset(double position_offset) const;
+    double getRescaleFactor() const;
+    void setRescaleFactor(double rescale_factor) const;
+    void resetFailsafe() const;
+    double getTargetPosition() const;
+    void setTargetPosition(double target_position) const;
+    double getVelocity() const;
+    double getVelocityLimit() const;
+    void setVelocityLimit(double velocity_limit) const;
+    double getMinVelocityLimit() const;
+    double getMaxVelocityLimit() const;
```

Comment:
While having this whole list of things implemented is nice, I'm generally of the opinion that we should only implement the bits that we need.  This reduces maintenance burden in the future; we don't have to change code that isn't there.  In this case, I would say that we should only implement what we are either setting or getting from `steppers_ros_i.cpp`; anything else is left for future expansion.

Reply:
I was hoping to use the min and max values to either coerce set values to be within the acceptable range or to provide a meaningful error message.

Reply:
Yeah, I'm totally fine with leaving what we are using.  So whatever you use, obviously keep.  Otherwise, we should just remove the bits we aren't using as it is effectively dead code at that point.